x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/ServiceUtils.java

```diff
@@ -283,6 +283,30 @@ public static Integer extractOptionalPositiveInteger(
         return optionalField;
     }
 
+    public static Float extractOptionalFloat(Map<String, Object> map, String settingName) {
```

Comment:
I might have missed them, but how about we add some unit tests for these new functions?